private/bufpkg/bufmodule/module_file_set.go

```diff
@@ -57,10 +57,27 @@ func newModuleFileSet(
 			),
 		)
 	}
+	multiReadBucket := newMultiModuleReadBucket(moduleReadBuckets...)
+	seenPaths := map[string][]string{}
+	if err := multiReadBucket.WalkModuleFiles(context.Background(), "", func(moduleObjectInfo *moduleObjectInfo) error {
+		modulePrefixedPaths, ok := seenPaths[moduleObjectInfo.Path()]
+		modulePrefixedPath := moduleObjectInfo.ExternalPath()
+		if moduleObjectInfo.ModuleIdentity() != nil {
+			modulePrefixedPath = moduleObjectInfo.ModuleIdentity().IdentityString() + "#" + modulePrefixedPath
```

Comment:
Let's discuss whether or not we should use a `#` here. With fully-qualified import paths in mind, we could probably just use `normalpath.Join` here so that the files are reported like they are in the BSR (e.g. https://buf.build/googleapis/googleapis/file/main/google/actions/type/date_range.proto minus the `file/main` components).

Reply:
Hmm, yeah, that's fair, I suppose `normalpath` is reasonable here since we want to represent the... well normalised path. My only fear with normalpath is that I don't want the path structure to confuse users that it's not necessarily a path on disk, but it should be fine. At least in general this check/change makes duplicated paths easier to debug.